validations.rb

```diff
@@ -0,0 +1,31 @@
+module Validations
+    def Validations.validate_amount(amount)
+        if amount > 0
+            return true
+        else
+            return false
+        end
+    end
+
+    def Validations.validate_withdrawal(amount, balance)
+        if amount > balance
+            return false
+        else
+            return true
+        end
+    end
+
+    def Validations.validate_withdrawal_daily_limit(amount, transactions)
+        daily_total = 0
+        transactions.each do |transaction|
+            if (transaction.date == Date.today && transaction.type == "withdrawal")
+                daily_total += transaction.amount
+            end
+        end
+        if daily_total + amount > 5000
+            return false
+        else
+            return true
+        end
+    end
+end
```

Comment:
I noticed that these validations always return true or false. This is an ok approach, but you could also do something like:
```ruby
def validate_something
  raise "Custom error message" if invalid
  # or even better
  raise StandardError.new "Custom error message" if invalid
end
```

after that, in main, you could do something like
```ruby
def do_something
  # logic to do something
rescue StandardError => e
  puts e # this will print "Custom error message"
end
```

Reply:
In specs, you can also do
```ruby
  expect { subject }.to raise_error(StandardError, "Custom error message")
```